source/compiler-core/slang-visual-studio-compiler-util.cpp

```diff
@@ -250,16 +252,34 @@ namespace Slang
     // Link options (parameters past /link go to linker)
     cmdLine.addArg("/link");
 
+    StringSlicePool libPathPool(StringSlicePool::Style::Default);
+
     for (const auto& libPath : options.libraryPaths)
     {
-        // Note that any escaping of the path is handled in the ProcessUtil::
-        cmdLine.addPrefixPathArg("/LIBPATH:", libPath);
+        libPathPool.add(libPath);
     }
 
     // Link libraries.
-    for (const auto& lib : options.libraries)
+    for (IArtifact* artifact : options.libraries)
+    {
+        auto desc = artifact->getDesc();
+
+        if (ArtifactInfoUtil::isCpuBinary(desc) && desc.kind == ArtifactKind::Library)
+        {
+            // Get the libray name and path
+            SLANG_RETURN_ON_FAIL(artifact->requireFileLike(ArtifactKeep::No));
+
+            libPathPool.add(ArtifactInfoUtil::getParentPath(artifact));
+            // We need the extension for windows
+            cmdLine.addArg(ArtifactInfoUtil::getBaseName(artifact) + ".lib");
```

Comment:
The last time I played around msvc command line here, splitting the path out in `/libpath:` section didn't not work. I'd make sure this code works for the heterogeneous-hello-world example.

Reply:
Can confirm the heterogeneous-hello-world example works.